internal/exportimport/handle_schedule.go

```diff
@@ -62,67 +63,77 @@ func (s *Server) handleSchedule() http.Handler {
 			}
 		}()
 
-		ctx, cancelFn := context.WithDeadline(ctx, time.Now().Add(s.config.MaxRuntime))
-		defer cancelFn()
-		logger.Info("starting export import scheduler")
+		ctx, cancel := context.WithDeadline(ctx, time.Now().Add(s.config.MaxRuntime))
+		defer cancel()
 
 		configs, err := s.exportImportDB.ActiveConfigs(ctx)
 		if err != nil {
-			logger.Errorw("unable to read active configs", "error", err)
+			logger.Errorw("failed to read active configs", "error", err)
+			s.h.RenderJSON(w, http.StatusInternalServerError, err)
 			return
 		}
 
-		httpClient := &http.Client{
-			Timeout: s.config.IndexFileDownloadTimeout,
-		}
-
-		anyErrors := false
-		for _, config := range configs {
-			logger.Infow("checking index file", "config", config)
-
-			req, err := http.NewRequestWithContext(ctx, "GET", config.IndexFile, nil)
-			if err != nil {
-				logger.Errorw("failed to create request to download index file", "file", config.IndexFile, "error", err)
-				continue
-			}
-
-			resp, err := httpClient.Do(req)
-			if err != nil {
-				anyErrors = true
-				logger.Errorw("error downloading index file", "file", config.IndexFile, "error", err)
-				continue
-			}
+		var merr *multierror.Error
 
-			defer resp.Body.Close()
```

Comment:
Twas bug. `defers` run when then function ends, not when the `for` loop ends. You could OOM for large numbers of schedules because you don't release the resources until everything has been processed. I moved the body into another function to ensure the body is released correctly.